### PR TITLE
fix: make the dispatch async; add await for create and edit annotation forms

### LINKED ISSUE #1230: Annotations: Correctly handle API errors
### Background
We currently display a success notification even if a request fails

### Reproduction Steps
1. Create an annotation
1. Edit the annotation to have more than 255 characters
   1. open the network tab
   1. save the edit
   1. observe the success message
   1. observe 400 failure network message

### Success Criteria
- API requests that fail have descriptive error messages

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #1230 

With our current state of handling the API call, we ship the call to a `dispatch` and this caused the error handling not possible because there was no catcher when the `dispatch` (making an `async` call) threw an error.

This PR makes that try-catch an async block and makes it so that a useful error message is shown on failure.

### CODE REVIEW

File: src/annotations/components/EditAnnotationOverlay.tsx [hunk 1]
```diff
@@ -33,15 +33,15 @@ export const EditAnnotationOverlay: FC = () => {
...
     try {
-      dispatch(editAnnotation(editedAnnotation))
+      await dispatch(editAnnotation(editedAnnotation))
       event('xyplot.annotations.edit_annotation.success')
       dispatch(notify(editAnnotationSuccess()))
       onClose()
     } catch (err) {
       event('xyplot.annotations.edit_annotation.failure')
-      dispatch(notify(editAnnotationFailed(err)))
+      dispatch(notify(editAnnotationFailed(err.response.data.message)))
```
Comment (drdelambre) [inline src/annotations/components/EditAnnotationOverlay.tsx:46]: there's no guarentee that the error caught using this method was created by an invalid fetch on the requested api. could have been caused by a code branch error anywhere in the editAnnotation action, or even at an auth layer on the api. 

something like `err.response?.data?.message || err` should be good enough to cover those edge cases

**Comment by drdelambre** [comment]
is it important that you're not waiting for this dispatch to complete? https://github.com/influxdata/ui/blob/master/src/annotations/actions/thunks.ts#L62

**Comment by ChitlangeSahas** [comment]
> is it important that you're not waiting for this dispatch to complete? https://github.com/influxdata/ui/blob/master/src/annotations/actions/thunks.ts#L62

From my understanding, the try catch is where we would have to wait if we want to catch the error for that block, I could be wrong tho.

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 64 | :yellow_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 87 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1293)

**Comment by drdelambre** [comment]
i was referring to the creator action on that line, but it seems like that's fully sync

**Review by drdelambre — APPROVED** [review]
seems fine as long as all the buttons work when you click on them

File: src/annotations/components/EditAnnotationOverlay.tsx (hunk 1 above)
Comment (Asalem1) [reply src/annotations/components/EditAnnotationOverlay.tsx]: There should be a utility function that's called `getErrorMessage` or something that could be useful in getting the relevant information

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 61 | :yellow_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 87 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1293)

File: src/annotations/components/EditAnnotationOverlay.tsx (hunk 1 above)
Comment (ChitlangeSahas) [reply src/annotations/components/EditAnnotationOverlay.tsx]: @Asalem1 can it live in the same `EditAnnotationOverlay.tsx` file?

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 87 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1293)

File: src/annotations/components/EditAnnotationOverlay.tsx (hunk 1 above)
Comment (Asalem1) [reply src/annotations/components/EditAnnotationOverlay.tsx]: High level, I don't see why not. It seems like this function should be a thunk that exists in a `thunks` file, but that's besides the point.

Comment (ChitlangeSahas) [reply src/annotations/components/EditAnnotationOverlay.tsx]: For sure, I'll add a function in the EditAnnotationOverlay.tsx file for now.

File: src/annotations/components/EditAnnotationOverlay.tsx [hunk 2]
```diff
@@ -33,15 +33,21 @@ export const EditAnnotationOverlay: FC = () => {
   const dispatch = useDispatch()
   const {clickedAnnotation} = useSelector(getOverlayParams)
 
-  const handleSubmit = (editedAnnotation: EditAnnotation): void => {
+  const getErrorMessage = (err: any): string | any => {
```
Comment (Asalem1) [inline src/annotations/components/EditAnnotationOverlay.tsx:36]: Not necessary, we can simply import this:

https://github.com/influxdata/ui/blob/master/src/utils/api.ts#L9

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 87 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1293)

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 59 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 87 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1293)

**Review by Asalem1 — APPROVED** [review]
LGTM

**Pull request merged**